routing/index.ts

```diff
@@ -0,0 +1,183 @@
+import {createAction} from '../actions';
+import {withTask} from '../tasks';
+import {HISTORY_PUSH_TASK} from '../history';
+
+type ComponentType = any;
+
+type RouteDefinition = {
+  [route: string]: ComponentType | [ComponentType, RouteDefinition]
+}
+
+/*
+ * Example usage:
+ *
+ * createRouter({
```

Comment:
Not really convinced by this api, this will be fine with simple configurations but might get quite difficult to read/maintain for larger apps.

In the same vibe, what's the purpose of the `ChildRoute` routing component, would it be a way to do something similar to the react-router api using jsx elements to construct routing? Having something like that would allow people to easily migrate, but maybe I misunderstood something.